File: Solved by Python/B. Napoleon Cake.py
# Nahin The Coder
# Date : 25 - 07 - 2021

test = int(input())
while test > 0:
    n = int(input())
    myList = list(map(int, input().strip().split()))[:n]
    temp = int(myList[n-1])
    i = int(n-1)
    while i >= 0:
        if myList[i] > 0:
            temp = max(myList[i], temp)
        if temp > 0:
            myList[i] = 1
            temp = temp - 1
        else:
            myList[i] = 0
        i = i-1
    print(*myList)
    test = test - 1

File: Part 2/B. Napoleon Cake.cpp
//Solver : Nahin the Coder
//Date : 25-07-2021

#include <bits/stdc++.h>
using namespace std;

int main() 
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    
    
    
    int test;
    cin>>test;
    while(test--)
    {
        int n;
        cin>>n;
        int arr[n];
        int arr2[n];
        
        for(int i=0; i<n; i++)
        {
            cin>>arr[i];
        }
        int temp = arr[n-1];
        for(int i=n-1; i>=0; i--)
        {
            if(arr[i]>0)
            {
                temp = max(arr[i],temp);
            }
            if(temp>0)
            {
                arr[i] = 1;
                temp--;
            }
            else
            {
                arr[i] = 0;
            }
        }
        for(int i=0; i<n; i++)
        {
            cout<<arr[i]<<" ";
        }
        cout<<endl;
    }
    return 0;
}
